File: 1302-DeleteCharactersToMakeFancyString/1302-DeleteCharactersToMakeFancyString.java
// Last updated: 10/20/2025, 11:40:05 AM
class Solution {
    public String makeFancyString(String s) {
        StringBuilder sb = new StringBuilder("");
        int cnt = 0;
        int idx = 0;
        char prev = ' ';
        for (char c : s.toCharArray()) {
            if (prev == c) {
                ++cnt;
            } else {
                cnt = 1;
                prev = c;
            }
            if (cnt < 3) {
                sb.append(c);
            }
            ++idx;
        }
        return new String(sb);
    }
}

File: 1302-DeleteCharactersToMakeFancyString/1302-DeleteCharactersToMakeFancyString.cpp
// Last updated: 10/20/2025, 11:40:09 AM
class Solution {
public:
    string makeFancyString(string s) {
        string res = "";
        res += s[0];
        int x = 0;
        for (int i = 1; i < s.size(); i++) {
            if (s[i] == s[i-1]) {
                ++x;
                if (x < 2) res += s[i];
                else continue;
            } else {
                x = 0;
                res += s[i];
            }
        }
        return res;
    }
};